docs/how-it-works/internals.md

```diff
@@ -0,0 +1,120 @@
+
+This experience informs the target user of Lonboard:
+
+- A Python user who does not know or want to know JavaScript.
+- Wants fast exploratory data analysis.
+- Already working with data in Python.
+- A reasonably fast internet connection on a desktop computer.
+
+### Python classes
+
+The `Map` class is the primary and only widget associated with JavaScript that renders anything. `Map` subclasses from `anywidget.AnyWidget`, which dynamically fetches Lonboard's JavaScript ESM bundle. The `Map` class emulates deck.gl's `Deck` class. It synchronizes map state and can be passed a sequence of `Layer` objects.
+
+`Layer` classes are designed to map to each underlying deck.gl layer. Most `Layer` classes subclass from `BaseArrowLayer`, which stores a `pyarrow.Table` in the dataclass. This data object gets serialized to Parquet and rendered in a layer provided by `@geoarrow/deck.gl-layers` as described later on. The primary API for each of these Arrow-based layers is `from_geopandas`, which converts a `GeoDataFrame` to a `pyarrow.Table` with a [GeoArrow](https://geoarrow.org/)-represented geometry column. A few layers are not Arrow-based and do not subclass from `BaseArrowLayer`: primarily `BitmapLayer` and `BitmapTileLayer`. These pass user data input directly into a core deck.gl layer.
+
+These `Layer` classes subclass from `ipywidgets.Widget` but are _not_ associated with any JavaScript of their own. But by being widgets themselves, their data is synced with JavaScript and Python data changes are propagated as events to the core `Map` object.
+
+Each layer has parameters that map to deck.gl accessors and render properties. Each of these has validation on the Python side to catch invalid input early. Accessors are modeled with ["trait" objects](https://developmentseed.org/lonboard/latest/api/traits/). Scalar accessors are stored as a scalar int/string/float. Function accessors are evaluated in Python and stored as a `pyarrow.Array`.
+
+A variety of [layer extensions](https://developmentseed.org/lonboard/latest/api/layer-extensions/) are supported, which map to deck.gl's upstream layer extensions. These layer extensions dynamically add more parameters to the layers on which they're assigned.
+
+
+### Data loading
+
+While pydeck allows users to pass in URLs to remote files, at the moment this is not possible in Lonboard. With only a few exceptions (e.g. the `BitmapLayer` and `BitmapTileLayer`), _only_ data originating in Python is supported in Lonboard. This reduction in scope vastly simplifies Lonboard's internals while not providing much of a hurdle for Lonboard's core audience.
```

Comment:
> only a few exceptions

This begs the question, why allow just these exceptions? They show that it is possible to pass through a URL and let deck "do the rest". Is the idea to add more in the future, or is there something special about `BitmapLayer`?

Reply:
As mentioned above, Lonboard is able to provide a good user experience with good performance because of the constraints it enforces on user input. This allows Lonboard to apply performance improvements under the hood without the user needing to learn anything special.

If we added a `GeoJsonLayer` which sent data over the wire as GeoJSON and used deck's `GeoJsonLayer`, then the user would have to understand the difference between that layer and the other layers, and we'd get issues about this layer being much smaller.

`BitmapLayer` is special because it doesn't have a direct GeoArrow counterpart.